File: source/direct3d9/MeshData.cpp
/*
* Copyright (c) 2007-2008 SlimDX Group
* 
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
* 
* The above copyright notice and this permission notice shall be included in
* all copies or substantial portions of the Software.
* 
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
* THE SOFTWARE.
*/
#include <d3d9.h>
#include <d3dx9.h>
#include <vcclr.h>
#include <memory>

#include "../ComObject.h"
#include "../DataStream.h"
#include "../Utilities.h"
#include "../StackAlloc.h"

#include "Device.h"
#include "Mesh.h"
#include "SkinInfo.h"
#include "ProgressiveMesh.h"
#include "MeshData.h"

using namespace System;

namespace SlimDX
{
namespace Direct3D9
{
	MeshData::MeshData( SlimDX::Direct3D9::Mesh^ mesh )
	{
		this->mesh = mesh;
		type = MeshDataType::Mesh;
	}

	MeshData::MeshData( SlimDX::Direct3D9::ProgressiveMesh^ mesh )
	{
		this->progressiveMesh = mesh;
		type = MeshDataType::ProgressiveMesh;
	}

	MeshData::MeshData( SlimDX::Direct3D9::PatchMesh^ mesh )
	{
		this->patchMesh = mesh;
		type = MeshDataType::PatchMesh;
	}

	void MeshData::Mesh::set( SlimDX::Direct3D9::Mesh^ value )
	{
		mesh = value;
		type = MeshDataType::Mesh;
	}

	void MeshData::ProgressiveMesh::set( SlimDX::Direct3D9::ProgressiveMesh^ value )
	{
		progressiveMesh = value;
		type = MeshDataType::ProgressiveMesh;
	}

	void MeshData::PatchMesh::set( SlimDX::Direct3D9::PatchMesh^ value )
	{
		patchMesh = value;
		type = MeshDataType::PatchMesh;
	}
}
}

File: tags/Sep 11/publish/SlimDX Samples (Sept 2011)/SampleFramework/Rendering/Mesh/MeshData.cs
/*
* Copyright (c) 2007-2011 SlimDX Group
* 
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
* 
* The above copyright notice and this permission notice shall be included in
* all copies or substantial portions of the Software.
* 
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
* THE SOFTWARE.
*/

using System.Collections.Generic;

namespace SlimDX.SampleFramework {
	/// <summary>
	/// A container for mesh data.
	/// </summary>
	public class MeshData {
		/// <summary>
		/// Gets position data.
		/// </summary>
		public List<Vector3> Positions {
			get {
				return positions;
			}	
		}

		/// <summary>
		/// Gets normal data.
		/// </summary>
		public List<Vector3> Normals {
			get {
				return normals;
			}
		}

		/// <summary>
		/// Gets texture coordinate data.
		/// </summary>
		public List<Vector2> TextureCoordinates {
			get {
				return textureCoordinates;
			}
		}
		
		/// <summary>
		/// Gets mesh index data.
		/// </summary>
		public List<int> Indices {
			get {
				return indices;
			}
		}
		
		List<Vector3> positions = new List<Vector3>();
		List<Vector3> normals = new List<Vector3>();
		List<Vector2> textureCoordinates = new List<Vector2>();
		List<int> indices = new List<int>();
	}
}
